src/app/core/cart/cart.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable, OnDestroy } from "@angular/core";
+import { fromEvent, Observable, Subscription } from "rxjs";
+
+@Injectable({
+    providedIn: 'root'
+})
+export class CartService implements OnDestroy {
+    private static readonly REFRESH_RATE: number = 1000;
+
+    private subscriptions: Subscription = new Subscription();
+    private snipcart: any = {};
+    cartTotal$: Observable<number> = new Observable();
+    cartWeight$: Observable<number> = new Observable();
+
+    constructor() {
+        this.subscriptions.add(
+            fromEvent(document, "snipcart.ready")
```

Comment:
This looks like the way to subscribe to this event. It definitely works, at least! Do you still want to make a service for this as you mentioned with `EventManager`, or is it better to just do this where we need to?

Reply:
I have implemented a solution, I will explain